src/rgw/rgw_admin.cc

```diff
@@ -5850,30 +5850,47 @@ int main(int argc, const char **argv)
   }
 
   if (opt_cmd == OPT_RESHARD_CANCEL) {
-    RGWReshard reshard(store);
-
     if (bucket_name.empty()) {
       cerr << "ERROR: bucket not specified" << std::endl;
       return EINVAL;
     }
-    cls_rgw_reshard_entry entry;
-    //entry.tenant = tenant;
-    entry.bucket_name = bucket_name;
-    //entry.bucket_id = bucket_id;
-    int ret = reshard.get(entry);
+
+    rgw_bucket bucket;
+    RGWBucketInfo bucket_info;
+    map<string, bufferlist> attrs;
+    ret = init_bucket(tenant, bucket_name, bucket_id, bucket_info, bucket, &attrs);
     if (ret < 0) {
-      cerr << "Error in getting bucket " << bucket_name << ": " << cpp_strerror(-ret) << std::endl;
-      return ret;
+      cerr << "ERROR: could not init bucket: " << cpp_strerror(-ret) << std::endl;
+      return -ret;
     }
 
-    /* TBD stop running resharding */
-
-    ret =reshard.remove(entry);
+    RGWBucketReshard br(store, bucket_info, attrs);
+    int ret = br.cancel();
     if (ret < 0) {
-      cerr << "Error removing bucket " << bucket_name << " for resharding queue: " << cpp_strerror(-ret) <<
+      cerr << "Error canceling bucket " << bucket_name << " resharding: " << cpp_strerror(-ret) <<
 	std::endl;
       return ret;
     }
+
+    RGWReshard reshard(store);
+
+    cls_rgw_reshard_entry entry;
+    //entry.tenant = tenant;
+    entry.bucket_name = bucket_name;
+    //entry.bucket_id = bucket_id;
+
+    ret = reshard.get(entry);
+    if (ret < 0 && ret != -ENOENT) {
+      cerr << "Error in getting bucket " << bucket_name << ": " << cpp_strerror(-ret) << std::endl;
+      return ret;
+    } else if (ret != -ENOENT) {
```

Comment:
@oritwas shouldn't we try to remove it even if it's -ENOENT?

Reply:
If get returned -ENOENT than it is not in the reshard log.
It is just how the code was written, there is no reason not to try to remove it and if we get -ENOENT ignore it.